lib/Dialect/ESI/Passes/ESIBuildManifest.cpp

```diff
@@ -57,8 +68,19 @@ void ESIBuildManifestPass::runOnOperation() {
   symCache.addDefinitions(mod);
   symCache.freeze();
 
-  // Gather the relevant types.
-  mod->walk([&](Operation *op) { scrapeTypes(op); });
+  // Find the top level appid hierarchy root.
+  for (auto root : mod->getRegion(0).front().getOps<AppIDHierRootOp>())
```

Comment:
`Operation *mod = getOperation();` -> `mlir::ModuleOp mod = getOperation();` since you anyways restricting pass scheduling at that level.
Then you don't need this `getRegion(0).front()` but can just refer to the body block.

Reply:
I'm trying to write code that is resilient to future changes in the outer container (the coming hw.design outer container).